app/src/main/java/com/sdp/movemeet/Activity/ListActivity.java

```diff
@@ -0,0 +1,102 @@
+import androidx.appcompat.app.ActionBarDrawerToggle;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+
+import com.google.android.material.navigation.NavigationView;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.sdp.movemeet.Backend.FirebaseInteraction;
+import com.sdp.movemeet.Navigation.Navigation;
+import com.sdp.movemeet.R;
+
+public class ListActivity extends AppCompatActivity {
+
+    TextView fullName, email, phone;
+    FirebaseAuth fAuth;
+    FirebaseFirestore fStore;
+    String userId;
+
+    DrawerLayout drawerLayout;
+    NavigationView navigationView;
+    Toolbar toolbar;
+    TextView textView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list);
+
+        createDrawer();
+        handleRegisterUser();
+    }
+
+    public void createDrawer() {
+        drawerLayout = findViewById(R.id.drawer_layout);
+        navigationView = findViewById(R.id.nav_view);
+        textView = findViewById(R.id.textView);
+        toolbar = findViewById(R.id.toolbar);
+
+        navigationView.bringToFront();
+        ActionBarDrawerToggle toggle = new
+                ActionBarDrawerToggle(this, drawerLayout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+        drawerLayout.addDrawerListener(toggle);
+
+        View hView = navigationView.inflateHeaderView(R.layout.header);
+
+        fullName = hView.findViewById(R.id.text_view_profile_name);
+        phone = hView.findViewById(R.id.text_view_profile_phone);
+        email = hView.findViewById(R.id.text_view_profile_email);
+
+        toggle.syncState();
+        navigationView.setNavigationItemSelectedListener(this::onNavigationItemSelected);
+        navigationView.setCheckedItem(R.id.nav_home);
+
+    }
+
+    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
+        switch (menuItem.getItemId()) {
+            case R.id.nav_home:
+                Navigation.goToHome(this.navigationView);
+                break;
+            case R.id.nav_edit_profile:
+                break;
+            case R.id.nav_add_activity:
+                Navigation.goToActivityUpload(this.navigationView);
+                break;
+            case R.id.nav_logout:
+                FirebaseInteraction.logoutIfUserNonNull(fAuth, this);
+                break;
+            case R.id.nav_start_activity:
+                Navigation.startActivity(this.navigationView);
+                break;
+            case R.id.nav_chat:
+                Navigation.goToChat(this.navigationView);
+                break;
+            case R.id.nav_list_activities:
+                Navigation.goToListOfActivities(this.navigationView);
```

Comment:
Is it really useful to make a redirection to this activities while we already are on the ListActivity ?

Reply:
You right, I remove it

Reply:
Also, @jlstoupy98, should this Navigation function include the `finish()` you mentioned for the `ActivityDescriptionActivity`?

Reply:
I add the finish in the last commit